bgpd/bgp_vty.c

```diff
@@ -10893,14 +10903,23 @@ DEFUN (show_ip_bgp_neighbors,
 	    || argv_find(argv, argc, "vrf", &idx))
 		vrf = argv[idx + 1]->arg;
 
+	if (argv_find(argv, argc, "ipv4", &idx)) {
```

Comment:
1. How does this command work when the issued command has both "vrf" & "ipv4 or ipv6" key words ?
2. Shouldn't this code be part of the old else loop present in the old code ?
3. Should idx++ be present before checking for ipv4 & ipv6 key words ?

Reply:
1. Parsing of the vrf argument is not affected with these changes. And hence vrf argument is passed as is to the bgp_show_neighbor_vty() along with ipv4, ipv6 types.
2. Didn't get this comment. Which line in the above changes are you referring to?
3. Actually idx is the starting index location for argv_find to search for the given keyword. We can still move it down, but it will not make any difference.

Reply:
#2: The old code has an else loop to handle when no IP address A.B.C.D is specified. i.e. sh_type = show_all. The new code added comes under the else loop actually. So my question is that, is there a specific reason for moving this before finding whether any IP address is specified or not ? The current code checks if ipv4/ipv6 key word is present or not, then also check for IP address. Are we supporting a case where user specifies ipv4/ipv6 key word and also IP address or IPv6 address ?
#3 Yes. It doesn't matter. My question is related to #2 above.

Reply:
Yes, user can specify both ipv4/ipv6 address family keyword and/or A.B.C.D/X:X.. address.

Reply:
Got it. Thanks.